js/base/Exchange.js

```diff
@@ -1631,4 +1630,16 @@ module.exports = class Exchange {
             throw new ExchangeError (this.id + ' this.twofa has not been set')
         }
     }
+
+    raiseError (exception, code = undefined, reason = undefined, url = undefined, method = undefined, headers = undefined, details = undefined) {
+        const ExceptionClass = exception.constructor
+        let errorString = [this.id, method, url, code, reason].filter (x => x !== undefined).join (' ')
```

Comment:
Can you make the error report look exactly as it does with the `verbose` mode on?

The existing verbose mode includes the request info and the response info.
The request info should contain the request detalis, including the body and the headers.
The response info should contain the response details, including the body and the headers.

Here's how a verbose-mode request-response pair should look like:

<img width="1136" alt="Screen Shot 2019-06-24 at 23 40 00" src="https://user-images.githubusercontent.com/1294454/60050544-8da54900-96d9-11e9-8b92-e553ee3c447a.png">

Here's how it looks like within this PR when the error occurs:

<img width="1136" alt="Screen Shot 2019-06-24 at 23 41 43" src="https://user-images.githubusercontent.com/1294454/60050609-add50800-96d9-11e9-88a6-fe3ede1526a0.png">

↑ It does contain the url, the http code and the reason, but the body occupies more than one screen, and it will be hard to copy-paste massive blocks into issues.

The best way to resolve this is to architect the error class itself properly. Namely, I suggest that we avoid `"\n"`-joining strings into the exception message, and we'd better have `class CCXTException` or `BaseError` with separate base fields:

```JavaScript
{
    url,
    httpMethod,
    httpCode,
    httpStatus,
    requestHeaders,
    requestBody,
    responseHeaders,
    responseBody,
    errorCode, // the error code defined by the exchange, if any
    message, // the error message defined by the exchange, if any
    exchangeId, // 
    exchangeMethod, // fetchOrders, fetchTickers, ...
    api, // (public, private)
}
```

I suggest that we rework the base exception class ↑ to accommodate the above.

The standard string serializers (`__str__`, `__repr__` and the analogies, in js, python and php) for that generalized error class would print out the verbose output looking close to the existing verbose output (readable).

Let me know your thoughts.

Reply:
Will take this into consideration as I try reattempt to make this PR. Not much point trying to build on this so I will close it and make a seperate PR with your suggestions. There are many advantages to doing it your way, including that users can catch the error and read properties directly from the error without having the parse the message.

Hope to get this done in the next few days...

Reply:
@frosty00 have you ever done another PR for this? Just wanted to reference in issue-response.